sdks/python/apache_beam/pipeline.py

```diff
@@ -221,6 +221,14 @@ def __init__(self, runner=None, options=None, argv=None):
     # then the transform will have to be cloned with a new label.
     self.applied_labels = set()  # type: Set[str]
 
+    # Create a context for assigning IDs to components. Ensures that any
+    # components that receive an ID during pipeline construction (for example in
+    # ExternalTransform), will receive the same component ID when generating the
+    # full pipeline proto.
+    from apache_beam.runners import pipeline_context
```

Comment:
There's not actually a circular dependency of classes. You could move this class in here and that would be fine.

Reply:
Done, thanks!